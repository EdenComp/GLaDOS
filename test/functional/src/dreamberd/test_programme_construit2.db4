fn display_info(name, age, temperature, isSummer) {
    print("Nom: " + name + "\n");
    print("Âge: " + age + "\n");
    print("Température: " + temperature + " degrés Celsius\n");

    if (isSummer) {
        print("C'est l'été!\n");
    } else {
        print("Ce n'est pas l'été.\n");
    }
}

fn calculate_discount(amount_spent) {
    var discount_rate = 0.1;

    if (amount_spent > 100.0) {
        return amount_spent * discount_rate;
    } else {
        return 0.0;
    }
}

fn calculate_age_in_months(age_in_years) {
    return age_in_years * 12;
}

fn main() {
    display_info("Alice", 25, 28.5, true);

    var total_amount = 120.0;
    var discount = calculate_discount(total_amount);

    print("Montant total: " + total_amount + "\n");
    print("Réduction applicable: " + discount + "\n");

    var age_in_months = calculate_age_in_months(25);
    print("Âge en mois: " + age_in_months + "\n");

    var age_sum = 25 + age_in_months;
    print("Somme de l'âge et de l'âge en mois: " + age_sum + "\n");
}

main();
