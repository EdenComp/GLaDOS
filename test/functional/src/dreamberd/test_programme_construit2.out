Nom: Alice
Âge: 25
Température: 28.5 degrés Celsius
C'est l'été!
Montant total: 120.0
Réduction applicable: 12.0
Âge en mois: 300
Somme de l'âge et de l'âge en mois: 325
